src/debug/PythonDebugProvider.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { DebugConfiguration, Extension, extensions, ShellExecution, ShellExecutionOptions, WorkspaceFolder } from 'vscode';
+import { convertToVenvCommand } from '../commands/createNewProject/PythonProjectCreator';
+import { funcHostStartCommand, hostStartTaskName, localhost } from '../constants';
+import { localize } from '../localize';
+import { getFuncExtensionSetting } from '../ProjectSettings';
+import { FuncDebugProviderBase } from './FuncDebugProviderBase';
+
+export const defaultPythonDebugPort: number = 9091;
+
+export const pythonDebugConfig: DebugConfiguration = {
+    name: localize('attachPython', 'Attach to Python Functions'),
+    type: 'python',
+    request: 'attach',
+    port: defaultPythonDebugPort,
+    preLaunchTask: hostStartTaskName
+};
+
+export class PythonDebugProvider extends FuncDebugProviderBase {
+    protected readonly defaultPort: number = defaultPythonDebugPort;
+    protected readonly debugConfig: DebugConfiguration = pythonDebugConfig;
+
+    public async getShellExecution(folder: WorkspaceFolder): Promise<ShellExecution> {
+        let command: string = funcHostStartCommand;
+        const port: number = this.getDebugPort(folder);
+        const venvName: string | undefined = getFuncExtensionSetting<string>('pythonVenv', folder.uri.fsPath);
+        if (venvName) {
+            command = convertToVenvCommand(venvName, process.platform, command);
+        }
+        const options: ShellExecutionOptions = { env: { languageWorkers__python__arguments: await getPythonCommand(localhost, port) } };
```

Comment:
Was the Python extension supposed to handle activating the virtual environment as well?  I thought that was part of the reason we wanted them to provide an API.